iroh-rpc-client/src/network.rs

```diff
@@ -257,6 +273,36 @@ impl P2pClient {
     }
 }
 
+#[derive(Debug)]
+pub struct Lookup {
+    pub peer_id: PeerId,
+    pub listen_addrs: Vec<Multiaddr>,
+    pub observed_addrs: Vec<Multiaddr>,
+    pub protocol_version: String,
+    pub agent_version: String,
+    pub protocols: Vec<String>,
+}
+
+impl Display for Lookup {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "peer id: {}\nlistening addresses: {:?}\nprotocols: {:?}\nprotocol version: {}\nobserved addresses: {:?}\n", self.peer_id, self.listen_addrs, self.protocols, self.protocol_version, self.observed_addrs)
```

Comment:
This is outputting vecs a `[a, b, c]`. Is that what we want to see in the CLI? I think nicer would be to use `.join` to create a comma separated list.

Reply:
handling this in a follow up